flask_static_digest/cli.py

```diff
@@ -1,21 +1,83 @@
-import click
+from dataclasses import dataclass
+from warnings import warn
 
+import click
 from flask import current_app
 from flask.cli import with_appcontext
 
-from flask_static_digest.digester import compile as _compile
 from flask_static_digest.digester import clean as _clean
+from flask_static_digest.digester import compile as _compile
+
+
+@dataclass
+class CompressionAlgo:
+    gzip: bool = False
+    brotli: bool = False
 
 
 @click.group()
-def digest():
-    """md5 tag and gzip static files."""
-    pass
+@click.pass_context
+@with_appcontext
+def digest(ctx):
```

Comment:
In my opinion we can drastically change this function.

- If we're going to use `ctx` then we should use it for `FLASK_STATIC_DIGEST_BLACKLIST_FILTER` too

I think we can mostly replace the whole function with what's below as well as delete the `CompressionAlgo` class. Keep in mind the code below isn't meant to be a complete solution. I wrote the code in this text box so the syntax might be off, it's meant to provide a general idea of what I was thinking:

```
def digest(ctx):
    ctx.obj = { gzip: False, brotli: False }

    ctx.obj.blacklist_filter = current_app.config.get("FLASK_STATIC_DIGEST_BLACKLIST_FILTER")

    compression = list(set(current_app.config.get("FLASK_STATIC_DIGEST_COMPRESSION")))

    for algo in compression:
       algo = algo.lower()

       if algo == "gzip":
           ctx.obj.gzip = True
       elif algo == "brotli"
           ctx.obj.brotli = True
       else:
          click.echo(f"{algo} is not a supported compression value, it must be 'gzip' or 'brotli'")
          sys.exit(1)
```

Then for the `compile` and `clean` functions, they can reference all 3 `ctx.obj` items as needed for their parameters.

Beyond that I don't think we need any more error handling. For example if someone tries to set `FLASK_STATIC_DIGEST_COMPRESSION` to be something other than a list or puts `True` for an algorithm item then the script will throw a Python exception.

If we go down this route of being hyper defensive we'll end up needing a lot of code and exception handling which I don't think is worth it considering this extension is run at build time. If it fails from any type of error the user will catch it then with a reasonably descriptive Python error that they can use as a guide.

Reply:
Pity, I like a nice dataclass as it enables the IDE to reason about it, autocomplete and prevents errors in the key names with dictionaries leading to wasted time debugging and any other annoyances. I would recommend this approach (and a class is effectively a dict anyway so runtime performance is good).

I am changing to your suggested dict , that is a very common Python pattern.